### DISCUSSION TITLE
Hitting Timeout on searches

### DISCUSSION THREAD

**Discussion by Bag72** [searxng/searxng#3223] [Q&A] [answered]
I'm sure I posted this before but I can't find it so I suspect I had posted in the wrong area and it was moderated out.  Apologies for that.

Installed SearXNG and it worked fine for months. Then it stopped working with the following failures

    Error! Engines cannot retrieve results:
    brave ( [timeout](https://search.petterson.network/stats?engine=brave) )
    duckduckgo ( [timeout](https://search.petterson.network/stats?engine=duckduckgo) )
    google ( [timeout](https://search.petterson.network/stats?engine=google) )
    qwant ( [timeout](https://search.petterson.network/stats?engine=qwant) )
    wikidata ( [timeout](https://search.petterson.network/stats?engine=wikidata) )
    wikipedia ( [timeout](https://search.petterson.network/stats?engine=wikipedia) )

I can ping each one from within the container without issue and there are no network issues at all.

I removed the containers and then installed again it worked fine. Just come back again and the error is back. Latest version of SearXNG in Docker.

Anyone else hitting that and any ideas how I can fix it?

**Comment by AC9G** [comment]
I had this same problem.  I'm trying these additional settings I found from:
https://github.com/searxng/searxng/issues/334

In the settings.yml I added:

outgoing:
  request_timeout: 5.0
  max_request_timeout: 10.0

I only had 1 timeout in the last week of trying this. So more tweaking necessary.

**Reply by Bag72** [reply]
OK.   I'll give that a go.  Thank you.

**Comment by return42** [comment, marked as answer]
> request_timeout: 5.0
> I only had 1 timeout in the last week of trying this

You should check the internet connectivity of your container (routing?) .. the default is 3sec and it works perfect for all SearXNG instances I know.

**Reply by Bag72** [reply]
So network issues.   I'll try the setting suppled by AC9G and see how that goes and if still problems I'll move the container to a different location and see how that goes.   I'm installing a NUC for services such as these and I'll be moving this there in the long run anyway.

Thank you both.

**Reply by Bag72** [reply]
Made the changes suggested by @AC9G and it worked great.  So far no issues at all in my setup so far.  Thank you.

@return42 I looked but don't see how to check the internet connectivity settings on Docker.   I'm fine with basic stuff but once it gets to the weeds I'm lost.     Networking is my Achilles heel probably because I don't sacrifice enough small furry animals to the dark gods or something.  :)
